cime_config/testdefs/testlist_clm.xml

```diff
@@ -2346,20 +2346,6 @@
       <option name="comment"  >Add at least one test of a FATES NEON site with PRISM precipitation</option>
     </options>
   </test>
-  <test name="SMS_D_Lm1_Mmpi-serial" grid="CLM_USRDAT" compset="I1PtClm50SpRs" testmods="clm/USUMB_nuopc">
```

Comment:
@slevis-lmwg I'd like you to reconsider this. The purpose of this one is to run a generic tower site and show that we can continue to do it. And that is a capability that we want to continue moving forward. NEON is highly curated so it doesn't provide a good example, and PLUMBER2 will be similar. But, we still want to show that someone can set something up for any random tower site.

Now, as I think about this, this site is setup to demonstrate PTCLM which we've removed. So it should be redone to use subset data and be based on it. That would be a good demonstration for this and for our current workflow.

I'll create an issue for this, and we can decide how this should happen. It wouldn't be hard to add to CTSM5.2, but it could also happen after CTSM5.2...

Reply:
Discussing how this should happen sounds good to me @ekluzek.

I'm encountering some other errors that I probably need your help with. I will start with these two:

1) The next two fail in MODEL_BUILD as "intel" but I expect them to work as "gnu" based on a third test that behaved the same way:
```
ERS_D_Ld5_Mmpi-serial.1x1_mexicocityMEX.I1PtClm51SpRs.derecho_intel.clm-CLM1PTStartDate
SMS_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm51FatesRs.derecho_intel.clm-FatesCold
```
The error says:
```
ld: /opt/cray/pe/mpich/8.1.25/ofi/intel/19.0/lib/libmpi_intel.so.12: undefined reference to `fi_strerror@FABRIC_1.0'
ld: /opt/cray/pe/mpich/8.1.25/ofi/intel/19.0/lib/libmpi_intel.so.12: undefined reference to `fi_fabric@FABRIC_1.1'
ld: /opt/cray/pe/mpich/8.1.25/ofi/intel/19.0/lib/libmpi_intel.so.12: undefined reference to `fi_getinfo@FABRIC_1.3'
ld: /opt/cray/pe/mpich/8.1.25/ofi/intel/19.0/lib/libmpi_intel.so.12: undefined reference to `fi_dupinfo@FABRIC_1.3'
ld: /opt/cray/pe/mpich/8.1.25/ofi/intel/19.0/lib/libmpi_intel.so.12: undefined reference to `fi_freeinfo@FABRIC_1.3'
ld: /opt/cray/pe/mpich/8.1.25/ofi/intel/19.0/lib/libmpi_intel.so.12: undefined reference to `fi_version@FABRIC_1.0'
```
In the first test, I see "fi" in 
`/glade/work/slevis/git/mksurfdata_toolchain/cime_config/testdefs/testmods_dirs/clm/CLM1PTStartDate/shell_commands`
Is that the "fi" that it's complaining about?

2) Then we have
```
FAIL SMS_Ld12_Mmpi-serial.1x1_urbanc_alpha.I1PtClm51SpRs.derecho_intel.clm-output_sp_highfreq RUN
```
failing with this error:
`Abort with message NetCDF: Variable not found in file /glade/derecho/scratch/csgteam/temp/spack/derecho/23.09/builds/spack-stage-parallelio-2.6.2-bpi7h2bnkshocep4hl3drfik4ebj44iu/spack-src/src/clib/pio_nc.c at line 1164`

Reply:
3. `SMS_Ld5.f09_g17.ISSP460Clm50BgcCrop.derecho_intel.clm-ciso_dec2050Start`
This one complains `ERROR: No stream_entry presaero.SSP4-6.0 found'`

Reply:
@slevis-lmwg the first one is due to: https://github.com/ESMCI/ccs_config_cesm/issues/130 So you just need to change it non-DEBUG or to gnu.

The second one (2) we need to see what file it's failing on. That might be in one of the log files, or you might see at least what file was last opening in the log files. Then we need to figure out what variable it is that it's failing to find...

On the third one (3) this means there likely isn't Prescribed aerosol files for SSP4-6.0. So we probably just shoudn't do any SSP4-6.0 tests. I would look into CDEPS to make sure this is a bug there though.

Reply:
I realize I'm coming to this kind of late, but regarding the single point test that @slevis-lmwg suggested removing the USUMB test created from PTCLM.  This does seem like something we should remove?

To avoid mission creep, I suggest we postpone testing for particular single point features to a PR that involves the development of single point features (NEON, PLUMBER2, or others), but defer to @ekluzek  and @slevis-lmwg, as you're more deeply involved here.

Reply:
Hey @wwieder yes that sounds great. I made this into an issue in #2356. I wasn't certain on when it should happen, but after CTSM5.2 is fine. It's also not hard to do, but makes sense to combine with other single points efforts and do it later...

Reply:
@ekluzek thank you:
1) Solved.
2) I updated the post (above) to include possible .nc files involved in the failure.
3) I opened an issue in CDEPS.

Next, I think found a bug in /ccs_config/component_grids_nuopc.xml. I had to change this line
```
-  <domain name="mpasa15-3conus">
+  <domain name="mpasa15-3">
```
for the case to find the grid's mesh file. Should I open another issue in CDEPS or does this go somewhere else?

Now I'm looking for PE layouts that may work for the ultra-high res grids. It's a very slow trial-and-error process due to derecho's slow turnaround (and my inexperience selecting PE layouts). However, now one of them is giving an error that appears reasonable
```
ERROR: reading in variable: TK_IMPROADERROR in /glade/work/slevis/git/mksurfdata_toolchain/src/main/ncdio_pio.F90.in at line 2261
```
I suspect this is still a memory issue, so I will keep increasing the number of nodes that I'm asking for.

Reply:
Submitted one more PE layout attempt for mpasa15, mpasa15-3, and mpasa3p75...

Reply:
Very exciting: Two of the three worked! The 3p75 case still failed.

Reply:
@ekluzek this far I've had no luck finding a PE layout for mpasa3p75. I have submitted another attempt to look at by morning. Meanwhile though, do you know if anyone in our community has run with this grid?

Reply:
Possibly @fischer-ncar? Chris, have you run mpasa3p75 on derecho successfully? If so, what PE layout would your recommend for a corresponding I-case?

Reply:
I think @fischer-ncar created this file before. It also looks like both you and I created a 16pft version earlier (me in October you in January). Is this for 15pft or 78pfts? 

The January version used 50 nodes. You might need to request both more nodes, and maybe high memory nodes, and/or ask for fewer tasks per node than 128.

Reply:
Oh, sorry I was thinking about file creation rather than running. I'll remove my comment above.

But, the same things apply. You likely need to give it a massive amount of processors. And asking for high memory nodes, and/or not using all processors on a node could apply here. We can't as easily change the number of tasks per node for a CTSM case. But, explicitly asking for memory can be done. Although I think we need to fiddle with env_batch.xml in your case for it...

Reply:
Failed attempts this far, listed here from less to more recent, as they appear in /glade/derecho/scratch/slevis/:
```
SMS_P128x50_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240209_140408_azgoww
SMS_P128x72_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240209_145631_lgb95u
SMS_P128x128_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240209_154145_jwj1mc
SMS_P128x256_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240209_170346_rq6ctr
SMS_P128x36_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240209_181654_97rslm
SMS_P36x128_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240209_194404_zontac
SMS_P72x128_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240212_105346_nbuwm1
SMS_P18x128_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240212_112748_x3v5ic
SMS_P256x128_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240212_140621_0ip7bj
SMS_P512x128_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240212_172315_eciutc
SMS_P1024x128_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240213_125603_7s8bgb
```
Next I attempt:
`SMS_P512x64_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240213_150027_87b1zr`

Reply:
@slevis-lmwg we should only use MPI tasks and NOT threading for this. There are currently problems with threading in the externals we are using. And in general threading is less efficient for CTSM than pure MPI. So only do tests with _P<tasks>x1_ .

For f09 we use 14 nodes on Derecho, 3.75 is about 1000X as many points. So using 14000 tasks is about right. And that's not that much bigger than the _P1024x128_ case. So I would try something like that. Which obviously might be hard to get in the batch queue...

Reply:
Oh, and 14,000 tasks doesn't evenly go into 128 processor nodes. So something like 14080 would be good to try.

Reply:
@ekluzek based on my understanding of your comment, I have now requested P14080x1. Does that look right?

If so, I also need to revisit tests that I got to work already, but for which I assigned funky PE layouts. Unless we should let them be, since they work.

Reply:
Yes, exactly P4080x1 that's right. We'll see what it does.

Reply:
Yeah, after you get the full list working in some fashion, let's look at the PE layouts together and see if they should be changed. We also might want to put some of these PE layouts into the config_pes.xml file rather than explicitly embedded into the testnames. But, getting something function is a huge step that we can refine later...

Reply:
The P14080x1 test failed, so I reached out to Brian D. to ask if there's a PE layout that has worked for him.

Reply:
Bummer. Yes, it will be good to hear from him.

What does the failure look like? Does it still look like memory issues? Can you point me to the failed case?

Reply:
Brian D. does not have a recommended PE layout for mpasa3p75, yet, and explained that the ultra high resolution work is very experimental. So I'm willing to stop spending time on this and labeling EXPECTED FAILURE for now. Thoughts @ekluzek and @wwieder ?

Reply:
@ekluzek `/glade/derecho/scratch/slevis/SMS_P14080x1_Ln9.mpasa3p75_mpasa3p75_mt13.I2000Clm51SpRs.derecho_intel.clm-clm51cam6LndTuningMode.G.20240213_164843_r7eiod/run`

Reply:
Ahhh, if it's experimental, yes we should remove it from our testing, and label it as expected failure. We shouldn't worry about testing it, until they can show us what to do. And I don't want to have a test that requires a 100 nodes of Derecho as just getting through the queue to run takes too long and is expensive.

Reply:
It's dying in malloc, so it's definitely a memory issue. So it's going to take some work to figure it out. It's at initialization so it might be initialization needs more memory than at run-time.

Reply:
@slevis-lmwg we should record this as an issue, so that people working on this can reference. And we'll be reminded of it being a problem. Could you start an issue for this?

Reply:
Yes.